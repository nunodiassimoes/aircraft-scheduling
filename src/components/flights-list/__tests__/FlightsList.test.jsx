import "@testing-library/jest-dom/extend-expect";
import { render, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { flightsFixture } from "utils/tests/fixtures";
import mockedAxios from "utils/tests/mockedAxios";

import FlightsList from "..";

const { mockAxios } = mockedAxios;

const renderFlightsList = () => {
  return render(<FlightsList />);
};

describe("FlightsList", () => {
  beforeEach(() => {
    jest.clearAllMocks();
    mockedAxios.resetAllMocks();
    mockedAxios.get.success(flightsFixture);
  });

  test("should display correctly the list of flights", async () => {
    const { container } = renderFlightsList();

    await waitFor(() => expect(mockAxios.get).toHaveBeenCalledTimes(1));
    expect(container).toMatchSnapshot();
  });

  test("should remove flight when adding it to the rotation list", async () => {
    const { getByTestId, queryByTestId } = renderFlightsList();

    userEvent.click(getByTestId("addFlightToRotationAS1001"));

    await waitFor(() =>
      expect(queryByTestId("flightsListItemAS1001")).toBeNull()
    );
  });
});
