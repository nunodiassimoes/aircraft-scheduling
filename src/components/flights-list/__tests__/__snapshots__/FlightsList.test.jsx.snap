// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlightsList should display correctly the list of flights 1`] = `
<div>
  <h2>
    Available flights
  </h2>
  <div
    class="infinite-scroll-component__outerdiv"
  >
    <div
      class="infinite-scroll-component "
      style="height: auto; overflow: auto;"
    >
      <div
        class="flightContainer"
        data-testid="flightsListItemAS1001"
      >
        <div
          class="id"
        >
          AS1001
        </div>
        <div
          class="col1_2"
        >
          <div>
            LFSB
          </div>
          <div>
            06:00
          </div>
        </div>
        <div
          class="col1_2"
        >
          <div>
            LFMN
          </div>
          <div>
            07:15
          </div>
        </div>
        <div>
          <button
            data-testid="addFlightToRotationAS1001"
          >
            Add flight to rotation
          </button>
        </div>
      </div>
      <div
        class="flightContainer"
        data-testid="flightsListItemAS1002"
      >
        <div
          class="id"
        >
          AS1002
        </div>
        <div
          class="col1_2"
        >
          <div>
            LFMN
          </div>
          <div>
            07:45
          </div>
        </div>
        <div
          class="col1_2"
        >
          <div>
            LFSB
          </div>
          <div>
            08:55
          </div>
        </div>
        <div>
          <button
            data-testid="addFlightToRotationAS1002"
          >
            Add flight to rotation
          </button>
        </div>
      </div>
      <div
        class="flightContainer"
        data-testid="flightsListItemAS1025"
      >
        <div
          class="id"
        >
          AS1025
        </div>
        <div
          class="col1_2"
        >
          <div>
            LFSB
          </div>
          <div>
            06:20
          </div>
        </div>
        <div
          class="col1_2"
        >
          <div>
            EDDH
          </div>
          <div>
            07:50
          </div>
        </div>
        <div>
          <button
            data-testid="addFlightToRotationAS1025"
          >
            Add flight to rotation
          </button>
        </div>
      </div>
      <div
        class="flightContainer"
        data-testid="flightsListItemAS1026"
      >
        <div
          class="id"
        >
          AS1026
        </div>
        <div
          class="col1_2"
        >
          <div>
            EDDH
          </div>
          <div>
            08:20
          </div>
        </div>
        <div
          class="col1_2"
        >
          <div>
            LFSB
          </div>
          <div>
            09:45
          </div>
        </div>
        <div>
          <button
            data-testid="addFlightToRotationAS1026"
          >
            Add flight to rotation
          </button>
        </div>
      </div>
      <div
        class="flightContainer"
        data-testid="flightsListItemAS1027"
      >
        <div
          class="id"
        >
          AS1027
        </div>
        <div
          class="col1_2"
        >
          <div>
            LFSB
          </div>
          <div>
            09:45
          </div>
        </div>
        <div
          class="col1_2"
        >
          <div>
            EDDH
          </div>
          <div>
            11:15
          </div>
        </div>
        <div>
          <button
            data-testid="addFlightToRotationAS1027"
          >
            Add flight to rotation
          </button>
        </div>
      </div>
      <div
        class="flightContainer"
        data-testid="flightsListItemAS1028"
      >
        <div
          class="id"
        >
          AS1028
        </div>
        <div
          class="col1_2"
        >
          <div>
            EDDH
          </div>
          <div>
            11:45
          </div>
        </div>
        <div
          class="col1_2"
        >
          <div>
            LFSB
          </div>
          <div>
            13:10
          </div>
        </div>
        <div>
          <button
            data-testid="addFlightToRotationAS1028"
          >
            Add flight to rotation
          </button>
        </div>
      </div>
      <div
        class="flightContainer"
        data-testid="flightsListItemAS1043"
      >
        <div
          class="id"
        >
          AS1043
        </div>
        <div
          class="col1_2"
        >
          <div>
            LFSB
          </div>
          <div>
            07:45
          </div>
        </div>
        <div
          class="col1_2"
        >
          <div>
            EHAM
          </div>
          <div>
            09:20
          </div>
        </div>
        <div>
          <button
            data-testid="addFlightToRotationAS1043"
          >
            Add flight to rotation
          </button>
        </div>
      </div>
      <div
        class="flightContainer"
        data-testid="flightsListItemAS1044"
      >
        <div
          class="id"
        >
          AS1044
        </div>
        <div
          class="col1_2"
        >
          <div>
            EHAM
          </div>
          <div>
            06:00
          </div>
        </div>
        <div
          class="col1_2"
        >
          <div>
            LFSB
          </div>
          <div>
            07:15
          </div>
        </div>
        <div>
          <button
            data-testid="addFlightToRotationAS1044"
          >
            Add flight to rotation
          </button>
        </div>
      </div>
      <div
        class="flightContainer"
        data-testid="flightsListItemAS1057"
      >
        <div
          class="id"
        >
          AS1057
        </div>
        <div
          class="col1_2"
        >
          <div>
            LFSB
          </div>
          <div>
            10:15
          </div>
        </div>
        <div
          class="col1_2"
        >
          <div>
            LEPA
          </div>
          <div>
            12:05
          </div>
        </div>
        <div>
          <button
            data-testid="addFlightToRotationAS1057"
          >
            Add flight to rotation
          </button>
        </div>
      </div>
      <div
        class="flightContainer"
        data-testid="flightsListItemAS1058"
      >
        <div
          class="id"
        >
          AS1058
        </div>
        <div
          class="col1_2"
        >
          <div>
            LEPA
          </div>
          <div>
            12:40
          </div>
        </div>
        <div
          class="col1_2"
        >
          <div>
            LFSB
          </div>
          <div>
            14:40
          </div>
        </div>
        <div>
          <button
            data-testid="addFlightToRotationAS1058"
          >
            Add flight to rotation
          </button>
        </div>
      </div>
      <div
        class="flightContainer"
        data-testid="flightsListItemAS1067"
      >
        <div
          class="id"
        >
          AS1067
        </div>
        <div
          class="col1_2"
        >
          <div>
            LFSB
          </div>
          <div>
            06:00
          </div>
        </div>
        <div
          class="col1_2"
        >
          <div>
            LEMG
          </div>
          <div>
            08:40
          </div>
        </div>
        <div>
          <button
            data-testid="addFlightToRotationAS1067"
          >
            Add flight to rotation
          </button>
        </div>
      </div>
      <div
        class="flightContainer"
        data-testid="flightsListItemAS1068"
      >
        <div
          class="id"
        >
          AS1068
        </div>
        <div
          class="col1_2"
        >
          <div>
            LEMG
          </div>
          <div>
            09:15
          </div>
        </div>
        <div
          class="col1_2"
        >
          <div>
            LFSB
          </div>
          <div>
            11:55
          </div>
        </div>
        <div>
          <button
            data-testid="addFlightToRotationAS1068"
          >
            Add flight to rotation
          </button>
        </div>
      </div>
      <div
        class="flightContainer"
        data-testid="flightsListItemAS1075"
      >
        <div
          class="id"
        >
          AS1075
        </div>
        <div
          class="col1_2"
        >
          <div>
            LFSB
          </div>
          <div>
            08:50
          </div>
        </div>
        <div
          class="col1_2"
        >
          <div>
            LEBL
          </div>
          <div>
            10:30
          </div>
        </div>
        <div>
          <button
            data-testid="addFlightToRotationAS1075"
          >
            Add flight to rotation
          </button>
        </div>
      </div>
      <div
        class="flightContainer"
        data-testid="flightsListItemAS1076"
      >
        <div
          class="id"
        >
          AS1076
        </div>
        <div
          class="col1_2"
        >
          <div>
            LEBL
          </div>
          <div>
            06:20
          </div>
        </div>
        <div
          class="col1_2"
        >
          <div>
            LFSB
          </div>
          <div>
            08:15
          </div>
        </div>
        <div>
          <button
            data-testid="addFlightToRotationAS1076"
          >
            Add flight to rotation
          </button>
        </div>
      </div>
      <div
        class="flightContainer"
        data-testid="flightsListItemAS1081"
      >
        <div
          class="id"
        >
          AS1081
        </div>
        <div
          class="col1_2"
        >
          <div>
            LFSB
          </div>
          <div>
            13:50
          </div>
        </div>
        <div
          class="col1_2"
        >
          <div>
            LEBL
          </div>
          <div>
            15:30
          </div>
        </div>
        <div>
          <button
            data-testid="addFlightToRotationAS1081"
          >
            Add flight to rotation
          </button>
        </div>
      </div>
      <div
        class="flightContainer"
        data-testid="flightsListItemAS1082"
      >
        <div
          class="id"
        >
          AS1082
        </div>
        <div
          class="col1_2"
        >
          <div>
            LEBL
          </div>
          <div>
            11:20
          </div>
        </div>
        <div
          class="col1_2"
        >
          <div>
            LFSB
          </div>
          <div>
            13:15
          </div>
        </div>
        <div>
          <button
            data-testid="addFlightToRotationAS1082"
          >
            Add flight to rotation
          </button>
        </div>
      </div>
      <div
        class="flightContainer"
        data-testid="flightsListItemAS1083"
      >
        <div
          class="id"
        >
          AS1083
        </div>
        <div
          class="col1_2"
        >
          <div>
            LFSB
          </div>
          <div>
            10:00
          </div>
        </div>
        <div
          class="col1_2"
        >
          <div>
            LEAL
          </div>
          <div>
            12:15
          </div>
        </div>
        <div>
          <button
            data-testid="addFlightToRotationAS1083"
          >
            Add flight to rotation
          </button>
        </div>
      </div>
      <div
        class="flightContainer"
        data-testid="flightsListItemAS1084"
      >
        <div
          class="id"
        >
          AS1084
        </div>
        <div
          class="col1_2"
        >
          <div>
            LEAL
          </div>
          <div>
            12:50
          </div>
        </div>
        <div
          class="col1_2"
        >
          <div>
            LFSB
          </div>
          <div>
            15:10
          </div>
        </div>
        <div>
          <button
            data-testid="addFlightToRotationAS1084"
          >
            Add flight to rotation
          </button>
        </div>
      </div>
      <div
        class="flightContainer"
        data-testid="flightsListItemAS11"
      >
        <div
          class="id"
        >
          AS11
        </div>
        <div
          class="col1_2"
        >
          <div>
            EGGW
          </div>
          <div>
            12:00
          </div>
        </div>
        <div
          class="col1_2"
        >
          <div>
            EGPH
          </div>
          <div>
            13:15
          </div>
        </div>
        <div>
          <button
            data-testid="addFlightToRotationAS11"
          >
            Add flight to rotation
          </button>
        </div>
      </div>
      <div
        class="flightContainer"
        data-testid="flightsListItemAS1115"
      >
        <div
          class="id"
        >
          AS1115
        </div>
        <div
          class="col1_2"
        >
          <div>
            LFSB
          </div>
          <div>
            06:00
          </div>
        </div>
        <div
          class="col1_2"
        >
          <div>
            LEMD
          </div>
          <div>
            08:20
          </div>
        </div>
        <div>
          <button
            data-testid="addFlightToRotationAS1115"
          >
            Add flight to rotation
          </button>
        </div>
      </div>
      <div
        class="flightContainer"
        data-testid="flightsListItemAS1116"
      >
        <div
          class="id"
        >
          AS1116
        </div>
        <div
          class="col1_2"
        >
          <div>
            LEMD
          </div>
          <div>
            08:55
          </div>
        </div>
        <div
          class="col1_2"
        >
          <div>
            LFSB
          </div>
          <div>
            11:10
          </div>
        </div>
        <div>
          <button
            data-testid="addFlightToRotationAS1116"
          >
            Add flight to rotation
          </button>
        </div>
      </div>
      <div
        class="flightContainer"
        data-testid="flightsListItemAS1125"
      >
        <div
          class="id"
        >
          AS1125
        </div>
        <div
          class="col1_2"
        >
          <div>
            LFSB
          </div>
          <div>
            12:05
          </div>
        </div>
        <div
          class="col1_2"
        >
          <div>
            LIRN
          </div>
          <div>
            13:45
          </div>
        </div>
        <div>
          <button
            data-testid="addFlightToRotationAS1125"
          >
            Add flight to rotation
          </button>
        </div>
      </div>
      <div
        class="flightContainer"
        data-testid="flightsListItemAS1126"
      >
        <div
          class="id"
        >
          AS1126
        </div>
        <div
          class="col1_2"
        >
          <div>
            LIRN
          </div>
          <div>
            09:45
          </div>
        </div>
        <div
          class="col1_2"
        >
          <div>
            LFSB
          </div>
          <div>
            11:35
          </div>
        </div>
        <div>
          <button
            data-testid="addFlightToRotationAS1126"
          >
            Add flight to rotation
          </button>
        </div>
      </div>
      <div
        class="flightContainer"
        data-testid="flightsListItemAS1131"
      >
        <div
          class="id"
        >
          AS1131
        </div>
        <div
          class="col1_2"
        >
          <div>
            LFSB
          </div>
          <div>
            10:05
          </div>
        </div>
        <div
          class="col1_2"
        >
          <div>
            LPPR
          </div>
          <div>
            12:40
          </div>
        </div>
        <div>
          <button
            data-testid="addFlightToRotationAS1131"
          >
            Add flight to rotation
          </button>
        </div>
      </div>
      <div
        class="flightContainer"
        data-testid="flightsListItemAS1132"
      >
        <div
          class="id"
        >
          AS1132
        </div>
        <div
          class="col1_2"
        >
          <div>
            LPPR
          </div>
          <div>
            07:05
          </div>
        </div>
        <div
          class="col1_2"
        >
          <div>
            LFSB
          </div>
          <div>
            09:30
          </div>
        </div>
        <div>
          <button
            data-testid="addFlightToRotationAS1132"
          >
            Add flight to rotation
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
